app/jobs/post_grouper.rb

```diff
@@ -0,0 +1,19 @@
+#class PostGrouper < ActiveJob::Base
+class PostGrouper
+  def perform(cleanup_gene_claims = true)
+    if cleanup_gene_claims
+      delete_orphaned_gene_claims()
+    end
+    update_counts()
+  end
+
+  def delete_orphaned_gene_claims
+      DataModel::GeneClaim.joins(:gene, :gene_claim_aliases, :source)
+          .where("gene_claims.gene_id IS NULL and sources.source_db_name = 'Ensembl'")
```

Comment:
Instead of looping over each claim and calling `destroy` which will generate `n` `delete from blah where id = ?` SQL queries, you could just do `Model.where().destroy_all` which will trigger a single `delete from blah where id in (list);`